src/devices/Mcp3xxx/Mcp3301.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Device.Spi;
+
+namespace Iot.Device.Adc
+{
+    /// <summary>
+    /// MCP3301 Analog to Digital Converter (ADC)
+    /// </summary>
+    public class Mcp3301 : Mcp3Base
+    {
+        /// <summary>
+        /// Constructs Mcp3301 instance
+        /// </summary>
+        /// <param name="spiDevice">Device used for SPI communication</param>
+        public Mcp3301(SpiDevice spiDevice) : base(spiDevice) { }
+
+        /// <summary>
+        /// Reads a 13-bit signed value from the device
+        /// </summary>
+        /// <returns>Integer value corresponding to relative voltage level on specified device channel</returns>
+        public int Read()
+        {
+            int retval = ReadInternal(adcRequest: 0, adcRequestLengthBytes: 2, adcResolutionBits: 13, delayBits: 0);
+
+            //convert 13 bit signed to 32 bit signed
+            return (retval >> 12) == 0 ? retval : (int)(0xFFFFE000 | retval);
```

Comment:
would it make sense to instead subtract `2^12` (+/- 1)? This is not exactly immediately obvious what's going on over here

Reply:
What I see here is that we are sign extending bit 13. What about if I just say that in the comment.

Reply:
O.K. I have created a static method for extending the sign... 

        /// <summary>
        /// Convert a signed value with a sign bit at a particular location to an int.
        /// </summary>
        /// <param name="signedValue">Signed value with a sign bit at a particular location</param>
        /// <param name="signingBit">Bit number that contains the sign bit</param>
        /// <returns>A value corresponding to the signed value sign extended into an int</returns>
        public static int SignExtend(int signedValue, int signingBit)
        {
            // if the sign bit is set then extend the signing bit to create a signed integer
            return (signedValue >> signingBit) == 0 ? signedValue : signedValue - (2 << signingBit);
        }